### ISSUE TITLE
Device not showing up in device page but showing up in Monitor tab

### ISSUE THREAD

**Issue by bmfmancini** [Cacti/cacti#3676]
Hey Guys 

I noticed that on 1.2.12 some devices are not showing up in the device search but in Monitor the device is there 
when you click on the device you are taken to its configuration page and its all filled out and even the graphs remain 

You are not able to delete the device since you cannot select it

**Comment by bmfmancini** [comment]
Here is the output from the DV 
I did poller sync as well no difference 

MariaDB [cacti]> select hostname,description,id,deleted,poller_id  from host where hostname='Device_IP';
+---------------+----------------+-------+---------+-----------+
| hostname      | description    | id    | deleted | poller_id |
+---------------+----------------+-------+---------+-----------+
| IP           | Device_hostname | 10788 |         |         7 |
+---------------+----------------+-------+---------+-----------+
1 row in set (0.00 sec)

**Comment by TheWitness** [comment]
Sort by id in the Cacti UI and look for that ID.  I don't think there are any joins there.  What's the status set to in the host table?

**Comment by TheWitness** [comment]
Also, check on that location bug we had in 1.2.12.  Could be a jacked up location setting.

**Comment by bmfmancini** [comment]
Hey @TheWitness 

I did a filter and to my surpirse it skips the ID in question !
Very strange

![image](https://user-images.githubusercontent.com/13388748/87269918-a1fee400-c49c-11ea-88d9-7f496204c006.png)

**Comment by TheWitness** [comment]
Edit host.php and at line ~1405 add the following:

```php
cacti_log(str_replace("\n", ' ', str_replace("\t", '', $sql_query)));
```

Post the result.

**Comment by bmfmancini** [comment]
Will do sometime between today and tomorrow 

Thanks !

**Comment by bmfmancini** [comment]
I just did that real quick in my Lab but it doesnt produce anything in the log 
this is where I put it  



                ))))) AS unsigned) AS instate
                FROM host
                LEFT JOIN (SELECT host_id, COUNT(*) AS graphs FROM graph_local GROUP BY host_id) AS gl
                ON host.id=gl.host_id
                LEFT JOIN (SELECT host_id, COUNT(*) AS data_sources FROM data_local GROUP BY host_id) AS dl
                ON host.id=dl.host_id
                $sql_where
                GROUP BY host.id
                $sql_order
                $sql_limit";

        return db_fetch_assoc($sql_query);
     cacti_log(str_replace("\n", ' ', str_replace("\t", '', $sql_query)));
}

**Comment by bmfmancini** [comment]
NVM I found out where I went wrong

**Comment by bmfmancini** [comment]
Here is the output

2020/07/13 15:48:09 - CMDPHP SELECT host.*, graphs, data_sources, CAST(IF(availability_method = 0, '0', IF(status_event_count > 0 AND status IN (1, 2), status_event_count*60, IF(UNIX_TIMESTAMP(status_rec_date) < 943916400 AND status IN (0, 3), total_polls*60, IF(UNIX_TIMESTAMP(status_rec_date) > 943916400, UNIX_TIMESTAMP() - UNIX_TIMESTAMP(status_rec_date), IF(snmp_sysUptimeInstance>0 AND snmp_version > 0, snmp_sysUptimeInstance/100, UNIX_TIMESTAMP() ))))) AS unsigned) AS instate FROM host LEFT JOIN (SELECT host_id, COUNT(*) AS graphs FROM graph_local GROUP BY host_id) AS gl ON host.id=gl.host_id LEFT JOIN (SELECT host_id, COUNT(*) AS data_sources FROM data_local GROUP BY host_id) AS dl ON host.id=dl.host_id WHERE (deleted = "" AND ( host.hostname LIKE '%10788%' OR host.description LIKE '%10788%' OR host.id = '10788')) AND IFNULL(host.location,"") = "" GROUP BY host.id ORDER BY `description` ASC LIMIT 0,30

**Comment by TheWitness** [comment]
Post that again with tripple backticks so I can see it raw.  When you run that on the MySQL line, do you not get any rows?

```sql
show create table host;
```

This filter is with location = "None" you should run:

```sql
SELECT location FROM host WHERE id = 10788;
```

**Comment by bmfmancini** [comment]
Fixed the comment 
I will check that and get back to you

**Comment by netniV** [comment]
Some good formatting there @bmfmancini  👍

**Comment by TheWitness** [comment]
This is the problem:

```sql
AND IFNULL(host.location,"") = ""
```

That will be for all hosts with no location, or location 'None', which on my system I don't see.

**Comment by TheWitness** [comment]
Okay, after a review of the code, it was obvious.  Glad to get it fixed.

**Comment by bmfmancini** [comment]
Hey @TheWitness  

Something is not right here 
During my testing all of the devices are disappearing now 



![captured (1)](https://user-images.githubusercontent.com/13388748/87369573-d632dd00-c54e-11ea-84d0-4adae58ba2b7.gif)

**Comment by TheWitness** [comment]
Yea, crap, not enough testing.  Pull host.php from https://github.com/TheWitness/cacti

From the 1.2.x branch.  The Cacti repo is still locked.

**Comment by bmfmancini** [comment]
Hey @TheWitness  tested the new commit looking good so far

**Comment by TheWitness** [comment]
Okay, well, there should always be a reason for a new release.  Love to get a perfectly clean one for sure.  Seems to be an elusive thing.

**Comment by bmfmancini** [comment]
Yes! totally agree!

Always glad to help testing the new fixes :)

**Comment by iskandarbasman** [comment]
Hi Guys,

You might want to a notice is that **host.php** is broken in Cacti 1.2.13.
Yeah all my devices disappear from Device Management.

Did a manual rollback of the code referencing Cacti 1.2.12.

Replaced this.

```
        if (get_request_var('location') == __('Undefined') || get_request_var('location') == '') {
                $sql_where .= ($sql_where != '' ? ' AND':' WHERE') . ' IFNULL(host.location,"") = ""';
        } elseif (get_request_var('location') != '') {
                $sql_where .= ($sql_where != '' ? ' AND':' WHERE') . ' host.location = ' . db_qstr(get_request_var('location'));;
```


**With This.**

```
        if (get_request_var('location') != '-1') {
                if (get_request_var('location') == __('Undefined') || get_request_var('location') == '') {
                        $sql_where .= ($sql_where != '' ? ' AND':' WHERE') . ' IFNULL(host.location,"") = ""';
                } else {
                        $sql_where .= ($sql_where != '' ? ' AND':' WHERE') . ' host.location = ' . db_qstr(get_request_var('location'));;
                }
```

**Comment by TheWitness** [comment]
It's fixed in the 1.2.x branch already.
